### DISCUSSION TITLE
Does not work for whtasapps from Brazil

### DISCUSSION THREAD

**Discussion by ferenczuk** [salman0ansari/whatsapp-api-nodejs#284] [General]
Awesome api, congratulations!
but I think it doesn't work for whatsapps in Brazil, because here there are whatsapp ids with 11 digits (nn nnnnn-nnnn) and others with 10 digits (nn nnnn-nnnn), but the mobile numbers are mandatory with 11 digits, so the end user will always put the id with 11 digits ( nn nnnnn-nnnn ) and thus not delivering the messages of whatssapp id is 10 digits.
Is there a way to configure this exception?

**Comment by AdrianoF0nseca** [comment]
ddd up to 27 -> 11 digits,
ddd from 28 to 99 -> 10 digits, it shouldn't be difficult to adapt this in your code

**Reply by ferenczuk** [reply]
Hi Adriano,
Thank you very much for your reply, problem solved.
Gratitude!

**Reply by JoaoNetoDev** [reply]
> Awesome api, congratulations! but I think it doesn't work for whatsapps in Brazil, because here there are whatsapp ids with 11 digits (nn nnnnn-nnnn) and others with 10 digits (nn nnnn-nnnn), but the mobile numbers are mandatory with 11 digits, so the end user will always put the id with 11 digits ( nn nnnnn-nnnn ) and thus not delivering the messages of whatssapp id is 10 digits. Is there a way to configure this exception?

In my solution for this I included the following function in "/src/api/class/instance.js":

async getWhatsAppJID(id) {
        const jidCorrect = await this.instance.sock?.onWhatsApp(id)
        return jidCorrect
    }

And I modified the onWhatsapp function in "/src/api/controllers/misc.controller.js":

exports.onWhatsapp = async (req, res) => {
    // eslint-disable-next-line no-unsafe-optional-chaining
    const data = await WhatsAppInstances[req.query.key]?.verifyId(
        WhatsAppInstances[req.query.key]?.getWhatsAppId(req.query.id)
    )
    const jidWpp = await WhatsAppInstances[req.query.key]?.getWhatsAppJID(req.query.id)
    return res.status(201).json({ error: false, data: data, jidWpp: jidWpp })
}

So, always before sending a message I first call "onWhatsapp", and what it returns in "JID" is what goes as "destination number", do you understand?
